packages/overlay/src/integrations/sentry/components/insights/FunctionProfiles.tsx

```diff
@@ -0,0 +1,111 @@
+
+  const functionProfiles = useMemo(() => {
+    const profiles = useSentryStore.getState().getFunctionProfiles();
+    const compareProfileInfo = COMPARATORS[sort.active] || COMPARATORS[FUNCTION_PROFILES_SORT_KEYS.timeSpent];
+
+    return profiles.sort((a, b) => {
+      return sort.asc ? compareProfileInfo(a, b) : compareProfileInfo(b, a);
+    });
+  }, [sort]);
+
+  if (!functionProfiles?.length) {
+    return <p className="text-primary-300 px-6 py-4">No profiles found.</p>;
+  }
+
+  // Calculate max time for bar visualization (100%, scaling form here)
+  const maxTime = Math.max(...functionProfiles.map(profile => profile.totalTime));
+
+  return (
+    <table className="divide-primary-700 w-full table-fixed divide-y">
+      <thead>
+        <tr>
+          {FUNCTION_PROFILES_HEADERS.map(header => (
+            <th
+              key={header.id}
+              scope="col"
+              className={classNames(
+                'text-primary-100 px-6 py-3.5 text-sm font-semibold',
+                header.primary ? 'w-2/5' : 'w-[15%]',
+              )}
+            >
+              <div
+                className={classNames(
+                  'flex cursor-pointer select-none items-center gap-1',
+                  header.primary ? 'justify-start' : 'justify-end',
+                )}
+                onClick={() => toggleSortOrder(header.sortKey)}
+              >
+                {header.title}
+                {sort.active === header.sortKey ? (
+                  <SortDown
+                    width={12}
+                    height={12}
+                    className={classNames(
+                      'fill-primary-300',
+                      sort.asc ? '-translate-y-0.5 rotate-0' : 'translate-y-0.5 rotate-180',
+                    )}
+                  />
+                ) : (
+                  <Sort width={12} height={12} className="stroke-primary-300" />
+                )}
+              </div>
+            </th>
+          ))}
+        </tr>
+      </thead>
+      <tbody>
+        {functionProfiles.map(profile => (
+          <tr key={`${profile.traceId}-${profile.name}`} className="hover:bg-primary-900">
+            <td className="text-primary-200 w-2/5 whitespace-nowrap px-6 py-4">
+              <TimeBar value={profile.totalTime} maxValue={maxTime} title={profile.name} />
+            </td>
+            <td className="text-primary-200 w-[15%] whitespace-nowrap px-6 py-4 text-right text-sm font-medium">
+              <span className={getSpanDurationClassName(profile.totalTime)}>
+                {getFormattedDuration(profile.totalTime)}
+              </span>
+            </td>
+            <td className="text-primary-200 w-[15%] whitespace-nowrap px-6 py-4 text-right text-sm font-medium">
+              {profile.samples}
+            </td>
+            <td className="text-primary-200 w-[15%] whitespace-nowrap px-6 py-4 text-right text-sm font-medium">
+              {profile.traceId.substring(0, 8)}
```

Comment:
We should probably create a `<TraceLink>` helper and standardize this (plus making it a link)